new/reclassify_amazon_linux_platform_family.md

```diff
@@ -0,0 +1,30 @@
+---
+RFC: unassigned
+Author: Sean OMeara <sean@chef.io>
+Status: Draft
+Type: Standards
+---
+
+# Title
+
+Reclassify Amazon Linux platform_family
+
+## Motivation
+
+   As time passes, Amazon Linux looks less and less like a RHEL clone.
+   Release version numbers do not match EL5, EL6, EL7.
+   Amazon is now a rolling release
+   Different package sets are available
+   Package names and contents are different
+   Service providers are different
+
+## Specification
+
+Change Amazon Linux platform_family from 'rhel' to 'amazon'
```

Comment:
In the case where you need to branch between fedora and amazon you have to branch anyway.  There's no net increase or decrease in work.  In the case where fedora and amazon match, however, then if they share the same platform_family then you can be lazy and you don't need to update your case statements and it'll just work.  With the 'amazon' platform_family approach now we have to through every cookbook in existence and update them to add `when "fedora", "amazon"` to every single fedora platform_family case.